dev/tools/controllerbuilder/template/controller/controller.go

```diff
@@ -254,17 +254,22 @@ func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error
 	if mapCtx.Err() != nil {
 		return nil, mapCtx.Err()
 	}
-	// TODO(user): Update other resource reference 
+	// TODO(user): Update other resource references 
 	parent, err := a.id.Parent()
 	if err != nil {
 		return nil, err
 	}
-	obj.Spec.ProjectRef = &refs.ProjectRef{Name: parent.ProjectID}
+	obj.Spec.ProjectRef = &refs.ProjectRef{External: parent.String()}
```

Comment:
@yuwenma regarding your question:

> what is the problem you hit ?

Some of these changes like setting external were review comments. Let's have the tooling do the right thing here.

Setting the name and gvk is the right thing to do for exporting.

Should I take out the labels for now?

Reply:
I think it was necessary to set name and GVK for marshaling SQLInstance when exporting: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/cae8025dee7e762f431bdb9a44946551b3edbc31/pkg/controller/direct/sql/sqlinstance_controller.go#L473-L474. It is probably also the right thing to set labels, though we don't have any tests for it yet: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/cae8025dee7e762f431bdb9a44946551b3edbc31/pkg/test/resourcefixture/contexts/register.go#L133-L136

Reply:
oh, didn't realize this is for Export. That's fair